eclair-core/src/main/scala/fr/acinq/eclair/io/PeerChannelsCollector.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2023 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.io
+
+import akka.actor.typed.scaladsl.adapter.TypedActorRefOps
+import akka.actor.typed.scaladsl.{ActorContext, Behaviors}
+import akka.actor.typed.{ActorRef, Behavior}
+import fr.acinq.bitcoin.scalacompat.Crypto.PublicKey
+import fr.acinq.eclair.Logs
+import fr.acinq.eclair.channel.{CMD_GET_CHANNEL_INFO, ChannelData, ChannelState, RES_GET_CHANNEL_INFO}
+
+import scala.concurrent.duration.FiniteDuration
+
+object PeerChannelsCollector {
+
+  // @formatter:off
+  sealed trait Command
+  case class GetChannels(replyTo: ActorRef[Peer.PeerChannels], channels: Set[akka.actor.ActorRef]) extends Command
+  private case class WrappedChannelInfo(state: ChannelState, data: ChannelData) extends Command
+  private object Timeout extends Command
+  // @formatter:on
+
+  /**
+   * @param timeout channel actors should respond immediately, so this actor should be very short-lived. However, if one
+   *                of the channel actors dies (because the channel is closed), we won't get a response: we will wait
+   *                until the given timeout, and then return the subset of channels we were able to collect.
+   */
+  def apply(remoteNodeId: PublicKey, timeout: FiniteDuration): Behavior[Command] = {
+    Behaviors.setup { context =>
+      Behaviors.withTimers { timers =>
+        Behaviors.withMdc(Logs.mdc(remoteNodeId_opt = Some(remoteNodeId))) {
+          Behaviors.receiveMessagePartial {
+            case GetChannels(replyTo, channels) =>
+              timers.startSingleTimer(Timeout, timeout)
```

Comment:
nit : this is simple and will work fine if the time-out is reasonable. the corner case described in the comment above (channel actor dies before it can reply) could also be handled using a death watch and a map indexed either by `ActorRef` (if they can indeed by used as map keys) or by associating random ids to each actor and adding them to the request and reply message.

Reply:
That's more satisfying indeed, I removed the timeout and watched child actors in https://github.com/ACINQ/eclair/pull/2688/commits/48f4dfad4af2e1e0e1e49929038544d57386c8d7